tests/unit/session/test_session.py

```diff
@@ -23,6 +24,29 @@
 from .. import resources
 
 
+@pytest.mark.parametrize(
+    ("engine", "write_engine"),
+    (
+        ("bigquery", "bigquery_streaming"),
+        ("bigquery", "bigquery_inline"),
```

Comment:
Add (None, "bigquery_external_table") too?

Reply:
Haven't implemented that one yet.

Reply:
Looks like we do throw the same exception from `validate_engine_compatibility`, which means this unit test extends to all these combinations:

({None, "c", "python", "pyarrow", "python-fwf"}, "bigquery_external_table")